eng/common/scripts/Helpers/Resource-Helpers.ps1

```diff
@@ -345,8 +345,9 @@ function RemoveStorageAccount($Account) {
     if ($container.BlobContainerProperties.HasImmutableStorageWithVersioning) {
       try {
         # Use AzRm cmdlet as deletion will only work through ARM with the immutability policies defined on the blobs
-        Remove-AzRmStorageContainer -Name $container.Name -StorageAccountName $Account.StorageAccountName -ResourceGroupName $Account.ResourceGroupName -Force
-        #$container | Remove-AzStorageContainer
+        # Add a retry in case blob deletion has not finished in time for container deletion, but not too many that we end up
+        # getting throttled by ARM/SRP if things are actually in a stuck state
+        Retry -Attempts 1 -Action { Remove-AzRmStorageContainer -Name $container.Name -StorageAccountName $Account.StorageAccountName -ResourceGroupName $Account.ResourceGroupName -Force }
```

Comment:
Why only retry 1 time?

Reply:
I don't feel 100% on this, but I think when I was trying to delete the container too quickly it would fail, as some blobs with immutability may have been cleaning up in the background. Adding the one retry gives us a 5 second sleep. However I don't want to retry multiple times, because if it is failing for valid reasons it just adds to our throttle count and we'll have that backoff/throttle problem all over again (especially if the failure itself is from throttling, we don't want to keep retrying too much).